pyccl/nl_pt/ept.py

```diff
@@ -0,0 +1,700 @@
+        return pim*self.exp_cutoff
+
+    def _get_pmm(self):
+        """ Get the one-loop matter power spectrum.
+
+        Returns:
+            array_like: 2D array of shape `(N_a, N_k)`, where `N_k` \
+                is the size of this object's `k_s` attribute, and \
+                `N_a` is the size of the object's `a_s` attribute.
+        """
+        self._check_pt_init()
+        if self.b1_pk_kind == 'linear':
+            P1loop = self._g4[:, None] * self.one_loop_dd[0][None, :]
+        else:
+            P1loop = 0.
+        pk = self.pk_b1 + P1loop
+        return pk*self.exp_cutoff
+
+    def get_pk2d_biased(self, tracer1, *, tracer2=None, return_ia_bb=False,
+                        extrap_order_lok=1, extrap_order_hik=2):
+        """Returns a :class:`~pyccl.pk2d.Pk2D` object containing
+        the PT power spectrum for two quantities defined by
+        two :class:`~pyccl.nl_pt.tracers.PTTracer` objects.
+
+        .. note:: The full non-linear model for the cross-correlation
+                  between number counts and intrinsic alignments is
+                  still work in progress in FastPT. As a workaround
+                  CCL assumes a non-linear treatment of IAs, but only
+                  linearly biased number counts.
+        Args:
+            tracer1 (:class:`~pyccl.nl_pt.tracers.PTTracer`): the first
+                tracer being correlated.
+            tracer2 (:class:`~pyccl.nl_pt.tracers.PTTracer`): the second
+                tracer being correlated. If `None`, the auto-correlation
+                of the first tracer will be returned.
+            return_ia_bb (bool): if `True`, the B-mode power spectrum
+                for intrinsic alignments will be returned (if both
+                input tracers are of type
+                :class:`~pyccl.nl_pt.tracers.PTIntrinsicAlignmentTracer`)
+                If `False` (default) E-mode power spectrum is returned.
+            extrap_order_lok (int): extrapolation order to be used on
+                k-values below the minimum of the splines. See
+                :class:`~pyccl.pk2d.Pk2D`.
+            extrap_order_hik (int): extrapolation order to be used on
+                k-values above the maximum of the splines. See
+                :class:`~pyccl.pk2d.Pk2D`.
+
+        Returns:
+            :class:`~pyccl.pk2d.Pk2D`: PT power spectrum.
+        """
+        if return_ia_bb:
+            return_ia_bb = True
+
+        if tracer2 is None:
+            tracer2 = tracer1
+
+        t1 = tracer1.type
+        t2 = tracer2.type
+
+        if ((t1 == 'NC') or (t2 == 'NC')) and (not self.with_NC):
+            raise ValueError("Need number counts Pk templates, "
+                             "but calculator didn't compute them")
+        if ((t1 == 'IA') or (t2 == 'IA')) and (not self.with_IA):
+            raise ValueError("Need intrinsic alignment Pk templates, "
+                             "but calculator didn't compute them")
```

Comment:
The error messages imply that the calculator is incompetent.
Better write `Can't use IA tracer in EulerianPTCalculator when 'with_IA=False'.` (but replace ' with `)

Reply:
Done

Reply:
Replace ' with `, because ' is reserved for strings, and it might confuse people.

Reply:
I disagree. The message will just read weird.

Reply:
We use ` I think throughout the code, but nevermind this is not worth it. Resolving.